cmd/maya-exporter/app/command/command.go

```diff
@@ -118,10 +121,23 @@ func Run(cmd *cobra.Command, options *VolumeExporterOptions) error {
 func (o *VolumeExporterOptions) RegisterJivaStatsExporter() error {
 	controllerURL, err := url.ParseRequestURI(o.ControllerAddress)
 	if err != nil {
-		log.Println(err)
+		glog.Error(err)
 		return errors.New("Error in parsing the URI")
 	}
-	exporter := collector.NewExporter(controllerURL)
+	exporter := collector.NewJivaStatsExporter(controllerURL)
 	prometheus.MustRegister(exporter)
 	return nil
 }
+
+// RegisterCstorStatsExporter initiates the connection with the cstor and register
+// the exporter with Prometheus for collecting the metrics.If the connection creation
+func (o *VolumeExporterOptions) RegisterCstorStatsExporter() {
+	var conn net.Conn
+	if conn = collector.InitiateConnection(); conn == nil {
+		glog.Error("Connection is not established with the target.")
+	}
```

Comment:
Shouldn't it return error from here?

Reply:
no, if it does not find connection it will set the conn = nil and retry to initiate connection.

Reply:
Please mention a comment in the code about why error should not be returned.